src/credentials-manager/index.js

```diff
@@ -0,0 +1,145 @@
+import {NativeModules} from 'react-native';
+import CredentialsManagerError from './credentialsManagerError';
+
+export default class CredentialsManager {
+  constructor(clientId, domain) {
+    this.domain = domain;
+    this.clientId = clientId;
+    this.Auth0Module = NativeModules.A0Auth0;
+  }
+
+  /**
+   * Saves the provided credentials
+   *
+   * @param {Object} credentials credential values
+   * @param {String} credentials.idToken required - JWT token that has user claims
+   * @param {String} credentials.accessToken required - token used for API calls
+   * @param {String} credentials.tokenType required - type of the token, ex - Bearer
+   * @param {Number} credentials.expiresIn optional - `expiresAt` should not be empty if this is. Used to denote when the token will expire from the issued time
+   * @param {String} credentials.expiresAt optional - `expiresIn` should not be empty if this is. Used to denote when the token will expire. Has precendence over `expiresIn`.
```

Comment:
Not sure on the docs here. Looking at the code, either `expiresIn` or `expiresAt` is required but at least one of them must be specified, but the docs say they're both optional.

Edit: ok I see now:

> credentials.expiresIn optional - `expiresAt` should not be empty if this is.

I found this really confusing and we should reword. How about:

"credentials.expiresIn - optional if `expiresAt` is set.."

And the converse for `expiresAt`.